parallel-crypto3/libs/parallel-zk/include/nil/crypto3/zk/commitments/detail/polynomial/basic_fri.hpp

```diff
@@ -496,21 +501,26 @@ namespace nil {
                 }
 
                 template<typename FRI, typename ContainerType,
-                        typename std::enable_if<
-                                std::is_base_of<
-                                        commitments::detail::basic_batched_fri<
-                                                typename FRI::field_type, typename FRI::merkle_tree_hash_type,
-                                                typename FRI::transcript_hash_type,
-                                                FRI::m, typename FRI::grinding_type>,
-                                        FRI>::value,
-                                bool>::type = true>
+                         typename std::enable_if<
+                             std::is_base_of<commitments::detail::basic_batched_fri<
+                                                 typename FRI::field_type,
+                                                 typename FRI::merkle_tree_hash_type,
+                                                 typename FRI::transcript_hash_type,
+                                                 FRI::m, typename FRI::grinding_type>,
+                                             FRI>::value,
+                             bool>::type = true>
                 static typename std::enable_if<
-                        (std::is_same<typename ContainerType::value_type, math::polynomial_dfs<typename FRI::field_type::value_type>>::value),
-                        typename FRI::precommitment_type>::type
-                precommit(ContainerType poly,
-                          std::shared_ptr<math::evaluation_domain<typename FRI::field_type>> D,
-                          const std::size_t fri_step
-                ) {
+                    (std::is_same<typename ContainerType::value_type,
```

Comment:
Maybe create some helper, like is_any_polynomial_dfs? Here and in another places?

Reply:
Changed.